exporters/otlp/otlpgrpc/connection.go

```diff
@@ -276,3 +282,127 @@ func (c *connection) contextWithStop(ctx context.Context) (context.Context, cont
 	}(ctx, cancel)
 	return ctx, cancel
 }
+
+func (c *connection) doRequest(ctx context.Context, fn func(context.Context) error) error {
+	expBackoff := newExponentialBackoff(c.cfg.RetrySettings)
+
+	for {
+		err := fn(ctx)
+		if err == nil {
+			// request succeeded.
+			return nil
+		}
+
+		if !c.cfg.RetrySettings.Enabled {
+			return err
+		}
+
+		// We have an error, check gRPC status code.
+		st := status.Convert(err)
+		if st.Code() == codes.OK {
+			// Not really an error, still success.
+			return nil
+		}
+
+		// Now, this is this a real error.
+
+		if !shouldRetry(st.Code()) {
+			// It is not a retryable error, we should not retry.
+			return err
+		}
+
+		// Need to retry.
+		var delay time.Duration
+
+		// Respect server throttling.
+		throttle := getThrottleDuration(st)
+
+		backoffDelay := expBackoff.NextBackOff()
+		if backoffDelay == backoff.Stop {
+			// throw away the batch
+			err = fmt.Errorf("max elapsed time expired %w", err)
```

Comment:
should it not be:

```suggestion
			err = fmt.Errorf("max elapsed time expired: %w", err)
```
?

reference: https://blog.golang.org/go1.13-errors